cortado-core/cortado/src/main/java/edu/utexas/cs/utopia/cortado/Driver.java

```diff
@@ -0,0 +1,643 @@
+ * <p>
+ * Command line interface:
+ * <p>
+ * Run
+ * java -jar lockPlacementBenchmarks.jar -h
+ */
+public class Driver
+{
+    private static final Logger log = LoggerFactory.getLogger(Driver.class);
+
+    /**
+     * Get index before first soot argument
+     *
+     * @param args command line args
+     * @return First index where "--" appears, or -1
+     */
+    private static int argsSplitIndex(String[] args)
+    {
+        for (int i = 0; i < args.length; ++i)
+            if (args[i].equals("--"))
+                return i;
+        return -1;
+    }
+
+    /**
+     * Run soot packages using soot command line args,
+     * then do our transforms
+     *
+     * @param args command line arguments
+     */
+    public static void main(String[] args) throws IOException, MaxSatSolverFailException, UnexpectedOutputException
+    {
+        // Separate cortado command line arguments from soot command line arguments
+        int splitIndex = argsSplitIndex(args);
+        String[] sootOptions, checkerOptions;
+
+        CmdLine cmdLine = new CmdLine();
+
+        if (splitIndex == -1)
+        {
+            checkerOptions = args;
+            sootOptions = new String[]{};
+        }
+        else
+        {
+            checkerOptions = Arrays.copyOfRange(args, 0, splitIndex);
+            sootOptions = Arrays.copyOfRange(args, splitIndex + 1, args.length);
+        }
+
+        // validate requested output format (and store requested output
+        // format in case the user wants bytecode, since CmdLine.setupSoot
+        // will force output to jimple)
+        Options.v().parse(sootOptions);
+        int cmdLineOutFormat = Options.v().output_format();
+        if (cmdLineOutFormat != Options.output_format_jimple && cmdLineOutFormat != Options.output_format_class)
+        {
+            System.err.println("Unsupported output format. Supported formats are jimple (-f jimple) and bytecode (default format)");
+            System.exit(1);
+        }
+
+        // Parse non-soot args
+        log.debug("Parsing non-soot command line arguments");
+        String err = cmdLine.parseArgs(checkerOptions);
+
+        if (err != null || cmdLine.isHelp())
+        {
+            if (err != null)
+                System.err.println(err);
+
+            System.err.println(cmdLine.usage());
+            System.exit(1);
+        }
+
+        // Create an object to log global statistics
+        CortadoProfiler.buildGlobalProfiler(cmdLine.getTargetClasses());
+        final CortadoProfiler cortadoProfiler = CortadoProfiler.getGlobalProfiler();
+        cmdLine.recordRunInfoToGlobalProfiler();
+        cortadoProfiler.startCortado();
+
+        // we'll use this to record the number of lockings, if it is
+        // counted. className -> list, list[i] = # of lockings using i locks
+        Map<String, List<Integer>> monitorToNumLockings = cmdLine.isEnumerateLockingsEnabled() ? new HashMap<>() : null;

```

Comment:
Can we actually remove the enumeration capabilities? It hasn't been tested in a while